torcs : geometry.cpp:373:15: error: 'isnan' was not declared in this scope

## Issue description
`torcs` failed to build with `geometry.cpp:373:15: error: 'isnan' was not declared in this scope` 

Full building log [torcs_log.txt](https://github.com/NixOS/nixpkgs/files/1404764/torcs_log.txt)

[patch](https://sourceforge.net/p/torcs/support-requests/11/) exists but I am not sure how to apply it correctly. Could be easy for anyone who know the process.

### Steps to reproduce
    $ cd /nix/nixpkgs/; git rev-parse HEAD
    6d86fcb86de75a307684cf2e6b8a518bd458427c

    $ nix-shell -I nixpkgs=/nix/nixpkgs -p torcs 2>&1 | tee torcs_log.txt

## Technical details

 * System: NixOS: 18.03.git.00e56fb (Impala)
 * Nix version: nix-env (Nix) 1.11.15
 * Nixpkgs version: "18.03pre117327.3fe7cddc30"
 * Sandboxing enabled: build-use-sandbox = true

--- a/pkgs/games/torcs/default.nix
+++ b/pkgs/games/torcs/default.nix
@@ -19,6 +19,10 @@ stdenv.mkDerivation rec {
       url = "https://anonscm.debian.org/git/pkg-games/torcs.git/plain/debian/patches/format-argument.patch";
       sha256 = "04advcx88yh23ww767iysydzhp370x7cqp2wf9hk2y1qvw7mxsja";
     })
+    (fetchpatch {
+      url = "https://anonscm.debian.org/git/pkg-games/torcs.git/plain/debian/patches/glibc-default-source.patch";
+      sha256 = "0k4hgpddnhv68mdc9ics7ah8q54j60g394d7zmcmzg6l3bjd9pyp";
+    })
   ];
 
   postPatch = ''

--- a/pkgs/games/torcs/default.nix
+++ b/pkgs/games/torcs/default.nix
@@ -15,6 +15,10 @@ stdenv.mkDerivation rec {
       url = "https://anonscm.debian.org/git/pkg-games/torcs.git/plain/debian/patches/gcc6-isnan.patch";
       sha256 = "16scmq30vwb8429ah9d4ws0v1w6ai59lvn7hcgnvfzyap42ry876";
     })
+    (fetchpatch {
+      url = "https://anonscm.debian.org/git/pkg-games/torcs.git/plain/debian/patches/format-argument.patch";
+      sha256 = "04advcx88yh23ww767iysydzhp370x7cqp2wf9hk2y1qvw7mxsja";
+    })
   ];
 
   postPatch = ''
@@ -27,8 +31,6 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ bash ];
 
   installTargets = "install datainstall";
-
-  hardeningDisable = [ "format" ];
 
   postInstall = ''
     wrapProgram $out/bin/torcs \

--- a/pkgs/games/torcs/default.nix
+++ b/pkgs/games/torcs/default.nix
@@ -1,4 +1,4 @@
-{ fetchurl, stdenv, mesa, freeglut, libX11, plib, openal, freealut, libXrandr, xproto,
+{ fetchpatch, fetchurl, stdenv, mesa, freeglut, libX11, plib, openal, freealut, libXrandr, xproto,
 libXext, libSM, libICE, libXi, libXt, libXrender, libXxf86vm, libvorbis,
 libpng, zlib, bash, makeWrapper }:
 
@@ -10,7 +10,14 @@ stdenv.mkDerivation rec {
     sha256 = "0kdq0sc7dsfzlr0ggbxggcbkivc6yp30nqwjwcaxg9295s3b06wa";
   };
 
-  patchPhase = ''
+  patches = [
+    (fetchpatch {
+      url = "https://anonscm.debian.org/git/pkg-games/torcs.git/plain/debian/patches/gcc6-isnan.patch";
+      sha256 = "16scmq30vwb8429ah9d4ws0v1w6ai59lvn7hcgnvfzyap42ry876";
+    })
+  ];
+
+  postPatch = ''
     sed -i -e s,/bin/bash,`type -P bash`, src/linux/torcs.in
   '';
 

